src/Explain.h

```diff
@@ -32,199 +36,206 @@
+                return true;
+            }
+            std::unique_ptr<TreeNode> t =
+                    prov.explainSubproof(query.first, label, ExplainConfig::getExplainConfig().depthLimit);
+            printTree(std::move(t));
+        } else if (command[0] == "explainnegation") {
+            std::pair<std::string, std::vector<std::string>> query;
+            if (command.size() == 2) {
+                query = parseTuple(command[1]);
+            } else {
+                printError(
+                        "Usage: explainnegation relation_name(\"<string element1>\", <number element2>, "
+                        "...)\n");
+                return true;
+            }
 
-                if (ncurses && !output) {
-                    prefresh(treePad, 0, 0, 0, 0, maxy - 3, maxx - 1);
-                }
+            size_t i = 1;
+            std::string rules;
+            for (auto rule : prov.getRules(query.first)) {
+                rules += std::to_string(i) + ": ";
+                rules += rule;
+                rules += "\n\n";
+                i++;
+            }
+            printInfo(rules);
 
-                printPrompt("Pick a rule number: ");
+            printPrompt("Pick a rule number: ");
 
-                std::string ruleNum = getInput();
-                auto variables =
-                        prov.explainNegationGetVariables(query.first, query.second, std::stoi(ruleNum));
+            std::string ruleNum = getInput();
+            auto variables = prov.explainNegationGetVariables(query.first, query.second, std::stoi(ruleNum));
 
-                if (variables.size() == 1 && variables[0] == "@") {
-                    printPrompt("The tuple exists, cannot explain negation of it!\n");
-                    continue;
-                } else if (variables.size() == 1 && variables[0] == "@non_matching") {
-                    printPrompt("The variable bindings don't match, cannot explain!\n");
-                    continue;
-                }
+            if (variables.size() == 1 && variables[0] == "@") {
+                printInfo("The tuple exists, cannot explain negation of it!\n");
+                return true;
+            } else if (variables.size() == 1 && variables[0] == "@non_matching") {
+                printInfo("The variable bindings don't match, cannot explain!\n");
+                return true;
+            }
 
-                // TODO (taipan-snake): this doesn't work with ncurses yet!!
-                std::map<std::string, std::string> varValues;
-                for (auto var : variables) {
-                    printPrompt("Pick a value for " + var + ": ");
-                    std::string varValue = getInput();
-                    varValues[var] = varValue;
-                }
+            std::map<std::string, std::string> varValues;
+            for (auto var : variables) {
+                printPrompt("Pick a value for " + var + ": ");
+                std::string varValue = getInput();
```

Comment:
Apart from helping debugging, I don't think anything is gained over the simpler `varValues[var] = getInput()`